### DISCUSSION TITLE
DataGrid ItemsSource Clear Err

### DISCUSSION THREAD

**Discussion by Nick-Suo** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#3516] [Q&A]
I write program in wpf6, When I use DataGrid to clear the content, I find that each line of the content will trigger an error, but the program does not crash.
I try in UI Clear ItemsSource or ViewModel Clear ,but this problem was not solved.
So I want to ask for help to help me solve this problem
The following is a screenshot of my code：
UI=> 
![image](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/assets/58356533/4a7e0e7e-3ba6-4204-b1b2-873000f8f904)
![image](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/assets/58356533/87e80dc6-c823-4695-a96a-e27561fbbef2)
ViewModel=>
![image](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/assets/58356533/8b37cae3-7a9d-4c8f-bd66-8b14306383da)

Error=>
![image](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/assets/58356533/eb818a4b-130e-4430-aa10-5251c5210a59)

**Comment by Keboo** [comment]
Difficult to say from just the screenshots. If you are able to provide a project that replicates the issue please open an issue and we can look into it further.

**Comment by Wil-fried** [comment]
Hello,

I'm also encountering theses errors. i don't have any projet that replicate the issue to provide (sorry) but i'm replying in hope to find a solution. I will try to describe the context as much as i can.

My solution use the materialDesignTheme (5.0.0) as Packages and i added the Generic.xaml (the doc/tuto recommand to use Defaults but it doesn't exist) and the MaterialDesign2.Defaults.xaml as resources which allow me to benefit from most of the userControl if i'm not wrong.
The mainwindow is composed of :
- a menu inspired by the one from the Demo that allow the switch between light and dark.
- a DataGrid fed by the defaultView of a DataTable, it have 250 columns and 100 rows. Hence why the DataGrid is configured with virtualization. 

Theses errors happen on runtime under the Xaml binding failure and i can't measure their impact on the app if there is any.
DataGrid itself seems to work fine if we leave performance apart. The only weird behavior i noticed is that everything change to Dark colors except the content of the DataGrid when switching to dark theme.
I found out that it's came from this part i added to catch doubleClick on a row :
```
<DataGrid.ItemContainerStyle>
   <Style TargetType="DataGridRow">
      <EventSetter Event="MouseDoubleClick" Handler="Row_DoubleClick"/>
   </Style>
</DataGrid.ItemContainerStyle>
```

In my case, all Xaml binding failure stop when i remove the column virtualization on the Datagrid
So here i'm! writing and editing this.
 Thank you a lot if you read me till this point.
